### PR TITLE
[Keymap] Added my own keymap for MIT Planck

### PR DESCRIPTION
**Pull request by jweickm**
## Description
It has a German and an English switching mode, so that it works the exact same with both OS languages.
It is a semi-wide layout, with the left hand all the way to the edge, and with one extra column in the middle used for media controls.
Home row mods implemented in the top row with individual tapping_terms for different fingers.
Mouse layer accessible through the pinkies.
Navigation layer on space.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/jweickm/config.h [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+#define WS2812_DMA_STREAM STM32_DMA1_STREAM2
+#define WS2812_DMA_CHANNEL 2
+
+#ifndef RGB_DISABLE_WHEN_USB_SUSPENDED
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED true
+#endif
+
+#endif
+
+#define RGBLIGHT_LAYERS
```
Comment (drashna) [inline keyboards/planck/keymaps/jweickm/config.h:150]:
```suggestion
```

File: keyboards/planck/keymaps/jweickm/keymap.c [hunk 2]
```diff
@@ -0,0 +1,1079 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "muse.h"
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/jweickm/keymap.c:19]:
```suggestion
```

File: keyboards/planck/keymaps/jweickm/keymap.c [hunk 3]
```diff
@@ -0,0 +1,1079 @@
...
+    return true;
+}
+
+bool     muse_mode      = false;
+uint8_t  last_muse_note = 0;
+uint16_t muse_counter   = 0;
+uint8_t  muse_offset    = 70;
+uint16_t muse_tempo     = 50;
+
+bool encoder_update(bool clockwise) {
```
Comment (drashna) [inline keyboards/planck/keymaps/jweickm/keymap.c:906]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/planck/keymaps/jweickm/rules.mk [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
+ENCODER_ENABLE = yes
+DIP_SWITCH_ENABLE = yes
+
+LAYOUTS = ortho_4x12 planck_mit
+LAYOUTS_HAS_RGB = no
```
Comment (drashna) [inline keyboards/planck/keymaps/jweickm/rules.mk:37]: You only need the features that you've changed here. 
```suggestion
TAP_DANCE_ENABLE = yes			# enable tap dance functionality
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/jweickm/rules.mk (hunk 4 above)
Comment (jweickm) [reply keyboards/planck/keymaps/jweickm/rules.mk:37]: Thank you for the feedback. It makes sense to remove the redundancy, so it always stays up to date. 
I still would like to keep `CONSOLE_ENABLE = no` as well, though, if that's okay.

File: keyboards/planck/keymaps/jweickm/keymap.c (hunk 3 above)
Comment (jweickm) [reply keyboards/planck/keymaps/jweickm/keymap.c:906]: I didn't realize that this had changed. Thanks for the update!

**Comment by jweickm** [comment]
@drashna 
Thanks again for the feedback. However, when I try to compile now after merging the changes, I get the following error regarding rgblighting
Do you have any idea what might cause this? (I'd like to avoid rerolling everything)
```  
keyboards/planck/keymaps/jweickm/keymap.c:1047:7: error: unknown type name 'rgblight_segment_t'
```

**Comment by jweickm** [comment]
> @drashna
> Thanks again for the feedback. However, when I try to compile now after merging the changes, I get the following error regarding rgblighting
> Do you have any idea what might cause this? (I'd like to avoid rerolling everything)
> 
> ```
> keyboards/planck/keymaps/jweickm/keymap.c:1047:7: error: unknown type name 'rgblight_segment_t'
> ```

Found the issue. The line 
`#define RGBLIGHT_LAYERS` was also

File: keyboards/planck/keymaps/jweickm/config.h [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#define RGBLIGHT_LAYERS
+
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/jweickm/config.h:19]:
```suggestion
#pragma once
```

File: keyboards/planck/keymaps/jweickm/config.h [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+//#define PERMISSIVE_HOLD
+#define IGNORE_MOD_TAP_INTERRUPT
+#define TAPPING_FORCE_HOLD_PER_KEY
+
+// settings for LEADER key
+#define LEADER_PER_KEY_TIMING
+#define LEADER_TIMEOUT 250
+#define LEADER_NO_TIMEOUT
+
+#define RGBLIGHT_SLEEP // RGB lighting will switch off when the host goes to sleep
```
Comment (drashna) [inline keyboards/planck/keymaps/jweickm/config.h:72]:
```suggestion
#define RGBLIGHT_SLEEP // RGB lighting will switch off when the host goes to sleep
#define RGBLIGHT_LAYERS
```

**Review by drashna — CHANGES_REQUESTED** [review]
Could you remove the submodule?

File: keyboards/planck/keymaps/jweickm/config.h (hunk 6 above)
Comment (jweickm) [reply keyboards/planck/keymaps/jweickm/config.h:72]: Thanks, that is a much better position for that line :)

**Comment by jweickm** [comment]
@drashna Submodule has been removed. I moved it to a different branch.

**Review by drashna — APPROVED** [review]

File: keyboards/planck/keymaps/jweickm/config.h [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+/* enable advanced MIDI features:
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
```
Comment (fauxpark) [inline keyboards/planck/keymaps/jweickm/config.h:53]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/jweickm/config.h (hunk 7 above)
Comment (jweickm) [reply keyboards/planck/keymaps/jweickm/config.h:53]: I somehow fail to see the difference between the two changes. You just remove the lines you added (or vice versa)?

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by jweickm** [comment]
Thank you guys for your time and valuable feedback!

**Pull request merged**